debug/debug_windows.go

```diff
@@ -0,0 +1,69 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+// +build windows
+
+package debug
+
+import (
+	"fmt"
+	"github.com/sirupsen/logrus"
+	"gopkg.in/natefinch/lumberjack.v2"
+	"path/filepath"
+	"time"
+)
+
+const (
+	maxSize = 15
+	maxBackups = 100
+	maxAge = 31
+	compress = true
+)
+
+func GetLumberjackLogger(file string) *lumberjack.Logger {
+	return &lumberjack.Logger{
+		Filename:   file,
+		MaxSize:    maxSize,
+		MaxBackups: maxBackups,
+		MaxAge:     maxAge,
+		Compress:   compress,
+	}
+}
+
+func SendEventsToLog(syslogIdentifier string, msg string, msgType string, delaySeconds time.Duration) {
+	filename := GetLogFilePath(syslogIdentifier)
+	logger := GetLumberjackLogger(filename)
+	logrus.SetOutput(logger)
+
+	switch msgType {
+	case "err":
+		logrus.Error(msg)
+	case "warning":
+		logrus.Warn(msg)
+	case "info":
+		logrus.Info(msg)
+	case "debug":
+		logrus.Debug(msg)
+	}
+	time.Sleep(delaySeconds * time.Second)
+}
+
+func GetLogFilePath(syslogIdentifier string) string {
+
+	logFileName := fmt.Sprintf("%s.log", syslogIdentifier)
+	logFile := filepath.Join( "C:/ProgramData", "Amazon", "fargate", "log", logFileName)
```

Comment:
Shim logger is open sourced and can potentially be used by other users. Should we still hard code `Amazon` and `fargate` in the log format?

Reply:
While we might be running the shim logger with Windows installed to `C:`, there's no guarantee that `%ProgramData%` will be on `C:`.  You can read the correct value with `os.Getenv("ProgramData")`.

Reply:
That is a good call out. I'll see what path I can use for it to work on all Windows envs.